### PR TITLE
collect custom lint rules from project

### PR DESCRIPTION
**Pull request by mohammadkahelghi-grabtaxi**
adapt bazel common collect custom lint rules from project feature
add transfering lint check modules to bazel targets

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -83,12 +87,17 @@ kotlin-serialization = { module = "org.jetbrains.kotlinx:kotlinx-serialization-j
...
+timber = { module = "com.jakewharton.timber:timber", version.ref = "timber" }
+auto-service = { module = "com.google.auto.service:auto-service", version.ref = "auto-service" }
 
 [plugins]
 kotlin-dsl = { id = "org.gradle.kotlin.kotlin-dsl", version.ref = "kotlin-dsl" }
 gradle-publish = { id = "com.gradle.plugin-publish", version.ref = "gradle-plugin-publish" }
-kotlin-serialization = { id = "org.jetbrains.kotlin.plugin.serialization", version.ref = "kotlin" }
\ No newline at end of file
+kotlin-serialization = { id = "org.jetbrains.kotlin.plugin.serialization", version.ref = "kotlin" }
+jetbrainsKotlinJvm = { id = "org.jetbrains.kotlin.jvm", version.ref = "jetbrains-kotlin-jvm" }
```
Comment (arunkumar9t2) [inline gradle/libs.versions.toml:103]: Please follow same pattern as others with `-`

File: grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/migrate/kotlin/KotlinProjectDataExtractor.kt [hunk 2]
```diff
@@ -66,19 +66,13 @@ constructor(
...
-            ).map { dependent ->
-                gradleDependencyToBazelDependency.map(project, dependent, null)
-            } +
-            dependenciesDataSource.collectMavenDeps(
-                project,
-                BuildGraphType(ConfigurationScope.BUILD)
-            ) +
-            project.androidJarDeps() +
-            project.kotlinParcelizeDeps()
+        val deps = projectDependencyGraphs.directDependencies(
```
Comment (arunkumar9t2) [inline grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/migrate/kotlin/KotlinProjectDataExtractor.kt:69]: Is this formatting correct? Default Kotlin formatting puts each params in separate line no?

File: sample-android/lint_baseline.xml [hunk 3]
```diff
@@ -1,6 +1,68 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <issues format="6" by="lint 8.5.0-alpha02" type="baseline" client="" dependencies="true" name="" variant="all" version="8.5.0-alpha02">
 
+    <issue
+        id="DuplicateDefinition"
```
Comment (arunkumar9t2) [inline sample-android/lint_baseline.xml:5]: I think this duplicate definition comes due to a cache issue, could you regenerate this baseline after a clean build and killing workers? I will investigate this cache issue separately.

Comment (mohammadkahelghi-grabtaxi) [reply sample-android/lint_baseline.xml:5]: sure

Comment (mohammadkahelghi-grabtaxi) [reply sample-android/lint_baseline.xml:5]: same result

File: grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/migrate/kotlin/KotlinProjectDataExtractor.kt [hunk 4]
```diff
@@ -97,27 +91,25 @@ constructor(
     fun lintConfigs(project: Project): LintConfigs {
```
[the remark this answers is not in the document]
Comment (mohammadkahelghi-grabtaxi) [reply grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/migrate/kotlin/KotlinProjectDataExtractor.kt]: moved

**Review by arunkumar9t2 — APPROVED** [review]

**Comment by arunkumar9t2** [comment]
Please merge after fixing the duplicate definition issue

**Pull request merged**